Removed the limitation on plan cache for partitioned tables. (#16359)
pingcap/docs: partitioned-table.md

From example 2, you can see that in `dynamic` mode, the execution plan with IndexJoin is selected when you execute the query.

> Currently, `static` pruning mode does not support plan cache, neither for prepared statmements nor non-prepared statements.

Reviewer: ```suggestion
Currently, `static` pruning mode does not support plan cache for both prepared and non-prepared statements.
```
